cmd/nri-kubernetes/main.go

```diff
@@ -244,14 +251,34 @@ func setupKubelet(c *config.Config, clients *clusterClients) (*kubelet.Scraper,
 		Kubelet:  clients.kubelet,
 		CAdvisor: clients.cAdvisor,
 	}
-	ksmScraper, err := kubelet.NewScraper(c, providers, kubelet.WithLogger(logger))
+
+	scraperOpts := []kubelet.ScraperOpt{kubelet.WithLogger(logger)}
+
+	if c.NamespaceSelector != nil {
+		scraperOpts = append(scraperOpts, kubelet.WithFilterer(buildNamespaceFilter(c, clients)))
+	}
+
+	ksmScraper, err := kubelet.NewScraper(c, providers, scraperOpts...)
 	if err != nil {
 		return nil, fmt.Errorf("building kubelet scraper: %w", err)
 	}
 
 	return ksmScraper, nil
 }
 
+func buildNamespaceFilter(c *config.Config, clients *clusterClients) *discovery.CachedNamespaceFilter {
+	nsFilter := discovery.NewNamespaceFilter(
+		c.NamespaceSelector,
+		clients.k8s,
+		logger,
+	)
+
+	return discovery.NewCachedNamespaceFilter(
+		nsFilter,
+		discovery.NewNamespaceInMemoryStore(c.Interval, logger),
```

Comment:
Probably I'd set this interval to be slightly smaller than the integration one, as I'd be good that the cache is cleaned before the new run.

Reply:
There's no TTL in the new cache, it just calls the vacuum on each interval.

Reply:
I've modified it with the new call to Vacuum, you can see the change in the last 2 commits.